grant-project/grant-monorepo
# Stubbed out comments UI

Closes #7.

### What This Does

This stubs out most of the UI for commenting. There's a lot to not like here that will hopefully get smoothed over by hooking up the API. But here's what we've got:

* Adds a comment box to the discussion tab that allows you to post a top level comment.
* Hooks up reply buttons to post a reply to a comment.
* Implements a distinction between "Full" markdown and "Reduced" markdown. Comments use reduced markdown, proposals (and in the future proposal updates) use full markdown.
* Implements somewhat of a message signing flow using a separate `signData` action provided by the web3 module. Presumably other sign-required actions could piggy-back in the same way. The UX could be improved here, but ideally it would only need to change in the one action, and anything that uses it for signing would be improved.
* (To be removed) does some local-state-only comment adding to redux state. This should be removed in favor of re-fetching the thread from the API, and re-rendering the whole thing.

### Things that need improving

* Comment form doesn't clear out when you submit it. This is partially due to not having the API hooked up.
* UI doesn't really reflect that you need to sign the message.
* Proposal ID and parent comment ID should probably be a part of the signed message as well.

### Steps to Test

* Go to a proposal, go to the discussion tab
* Attempt to post a comment, confirm you have to sign
* Cancel signature, confirm comment doesn’t post
* Accept signature, confirm comment posts
* Reply to comment, confirm it works as expected

## Screenshots

<img width="847" alt="screen shot 2018-09-17 at 3 38 56 pm" src="https://user-images.githubusercontent.com/649992/45647871-fb5e7d80-ba94-11e8-8f53-bd043b0565da.png">

---

<img width="472" alt="screen shot 2018-09-17 at 2 45 04 pm" src="https://user-images.githubusercontent.com/649992/45647865-f8638d00-ba94-11e8-8412-20e3ba640e7b.png">
